ateam_vision_filter/src/filters/multiple_hypothesis_tracker.cpp

```diff
@@ -60,105 +57,40 @@ void MultipleHypothesisTracker::update(const std::vector<Eigen::VectorXd> & meas
   // The left set (a,b,c) and right set (1,2,3) is fully connected
   // All edges have 1 flow representing 1 to 1 assignment of measurements to tracks
   // The cost between the left set and right set represent the distance between
-
-  source = boost::add_vertex(graph);
-  sink = boost::add_vertex(graph);
-
-  std::map<adjacency_list_traits::vertex_descriptor,
-    const Eigen::VectorXd &> vertex_to_measurement;
-  for (size_t i = 0; i < measurements.size(); i++) {
-    vertex_to_measurement.insert({boost::add_vertex(graph), measurements.at(i)});
-  }
-
-  std::map<adjacency_list_traits::vertex_descriptor,
-    InteractingMultipleModelFilter &> vertex_to_track;
-  for (size_t i = 0; i < tracks.size(); i++) {
-    vertex_to_track.insert({boost::add_vertex(graph), tracks.at(i)});
-  }
-
-  forward_edge_to_vertex_pair_t forward_edge_to_vertex_pair;
-
-  // Add all source to first set
-  for (auto & vertex_measurement_pair : vertex_to_measurement) {
-    add_edge_to_graph(
-      graph,
-      edge_capacity_list,
-      edge_weight_list,
-      edge_reverse_list,
-      forward_edge_to_vertex_pair,
-      source,
-      vertex_measurement_pair.first,
-      0.0);
-  }
-
-  // Add all second set to sink
-  for (auto & vertex_track_pair : vertex_to_track) {
-    add_edge_to_graph(
-      graph,
-      edge_capacity_list,
-      edge_weight_list,
-      edge_reverse_list,
-      forward_edge_to_vertex_pair,
-      vertex_track_pair.first,
-      sink,
-      0.0);
-  }
-
-  // We don't actually care about the source -> * and * -> sink so just clear edge to vertex map
-  forward_edge_to_vertex_pair.clear();
-
-  // Add cost for every single combination of measurement to track
-  for (const auto & vertex_measurement_pair : vertex_to_measurement) {
-    for (auto & vertex_track_pair : vertex_to_track) {
-      double dist =
-        (vertex_measurement_pair.second - vertex_track_pair.second.get_position_estimate()).norm();
-      add_edge_to_graph(
-        graph,
-        edge_capacity_list,
-        edge_weight_list,
-        edge_reverse_list,
-        forward_edge_to_vertex_pair,
-        vertex_measurement_pair.first,
-        vertex_track_pair.first,
-        dist);
+  // setup costs
+
+    Eigen::MatrixXd costs = Eigen::MatrixXd::Constant(tracks.size(), measurements.size(), std::numeric_limits<double>::infinity());
+    for (size_t i = 0; i < tracks.size(); i++) {
+        for (size_t j = 0; j < measurements.size(); j++) {
+            double dist = (measurements.at(j) - tracks.at(i).get_position_estimate()).norm();
+            costs(i, j) = dist;

```

Comment:
Do we need an intermediate variable here?

```suggestion
            costs(i, j) = (measurements.at(j) - tracks.at(i).get_position_estimate()).norm();

```